### ISSUE TITLE
[Camera] warning architecture on Xcode.

### ISSUE THREAD

**Issue by tonypottera24** [flutter/flutter#61927]
## Steps to Reproduce

`camera: ^0.5.8+2`
1. Run `flutter run -v` on iOS simulator `iPhone 11 iOS 13.6`.

**Expected results:** 
no warning

**Actual results:** 
`warning: Mapping architecture armv7 to i386`
`warning: Mapping architecture arm64 to x86_64`

<details>
  <summary>Logs</summary>

```
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: Mapping architecture armv7 to i386. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'camera' from project 'Pods')
    warning: Mapping architecture arm64 to x86_64. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'camera' from project 'Pods')
```

```
[✓] Flutter (Channel stable, v1.17.5, on Mac OS X 10.15.6 19G73, locale en-US)
    • Flutter version 1.17.5 at /Users/x/.flutter_stable
    • Framework revision 8af6b2f038 (3 weeks ago), 2020-06-30 12:53:55 -0700
    • Engine revision ee76268252
    • Dart version 2.8.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0-rc2)
    • Android SDK at /Users/x/Library/Android/sdk
    • Platform android-29, build-tools 30.0.0-rc2
    • ANDROID_HOME = /Users/x/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.8.4

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.47.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.2

[✓] Connected device (1 available)
    • iPhone 11 •• ios • com.apple.CoreSimulator.SimRuntime.iOS-13-6 (simulator)

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
This happens on latest Flutter dev channel and with latest `camera` plugin.  Tested on iphone simulator `iPhone 11 (13.5)` and can see the warnings in the log:


```
warning: Mapping architecture armv7 to i386. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'camera' from project 'Pods')

warning: Mapping architecture arm64 to x86_64. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'camera' from project 'Pods')
```
Digging more into the warnings, I noticed that these warnings don't occur if `deployment target` is set to `9.0`, but occurs on `8.0`.

Also, the `camera` plugin doesn't specify arch settings for `iphone simulator` in their `podspec`, as below:

https://github.com/flutter/plugins/blob/master/packages/camera/ios/camera.podspec

`s.pod_target_xcconfig = { 'DEFINES_MODULE' => 'YES', 'VALID_ARCHS' => 'armv7 arm64 x86_64' }`

whereas, other plugin, such as `video_player` uses valid arch settings in their `podspec`, as below:

https://github.com/flutter/plugins/blob/486da06ded94a169d98c5f9c18ca3aad91b4c501/packages/video_player/video_player/ios/video_player.podspec#L22

`s.pod_target_xcconfig = { 'DEFINES_MODULE' => 'YES', 'VALID_ARCHS[sdk=iphonesimulator*]' => 'x86_64' }`

**Comment by jmagman** [comment]
This should be fixed in https://github.com/flutter/plugins/pull/2928 and is available in camera 0.5.8+5.
